frontend/src/pages/Match/index.js

```diff
@@ -41,10 +43,13 @@ function MatchPage() {
   // useEffect
   // Difficulty
   useEffect(() => {
-    if (location.state) {
-      setDifficulty(location.state?.difficulty);
+    if (location.state?.difficulty) {
```

Comment:
Can I clarify when this case (user not selecting a difficulty and they are navigated to the waiting page) will happen?

Reply:
> 2. A useless match request, i.e., without the required difficulty argument, is emitted when user explicitly types the match page URL

Can refer to [Issue#40](https://github.com/CS3219-AY2223S1/cs3219-project-ay2223s1-g28/issues/40) for a demo.